packages/next/build/jest/jest.ts

```diff
@@ -38,6 +38,11 @@ function setUpEnv(dir: string, nextConfig: NextConfigComplete) {
   }
 }
 
+export type NextJestOptions = {
+  dir?: string
+  experimentalEsm?: boolean
```

Comment:
In conversation with @ijjk, they suggested `experimentalProcessNodeModules`, but I feel like that naming kind of obscures what this intends to solve, which is, if you're using ESM, you'll want to have `node_modules` transformed so Jest can load them, until Jest has proper ESM support.

Reply:
Yeah may be fine to be specific to ESM although having different naming from `experimental-esm` might be good to prevent confusion with Jest's experimental ESM support x-ref: https://jestjs.io/docs/ecmascript-modules

Reply:
perhaps it'd be better as:

```typescript
export type NextJestOptions = {
  dir?: string
  experimental?: {
    transformEsmModules?: boolean
  }
}
```

So if `experimental.transformEsmModules === true` then `/node_modules/` won't be in the transformIgnorePatterns array.